### PR TITLE
Add a convenience method for extracing the root API URL

### PR DESCRIPTION
**Pull request by zachmullen**

### CODE REVIEW

**Comment by zachmullen** [comment]
This is a bit hackish, but it should work for us. We can always change the implementation later if we come up with something that feels better.

File: girder/api/rest.py [hunk 1]
```diff
@@ -38,6 +38,23 @@
...
+    The returned path does *not* end in a forward slash.
+
+    :param url: URL from which to extract the base URL. If not specified, uses
+        `cherrypy.url()`
+    """
+    url = url or cherrypy.url()
+    idx = url.find('/api/v1')
+
+    if idx < 0:
+        raise Exception('Could not determine API root in %s.' % url)
```
Comment (cpatrick) [inline girder/api/rest.py]: This should be a subclass of Exception.

**Comment by zachmullen** [comment]
@cpatrick

**Comment by cpatrick** [comment]
LGTM :shipit:

**Pull request merged**